frontend-svelte/src/routes/new_challenge/[slug]/python-runner.ts

```diff
@@ -0,0 +1,114 @@
+import { writable } from 'svelte/store';
+
+const timer = ms => new Promise(res => setTimeout(res, ms))
+// Since pyodide is included via CDN we can't just use it right away, it might not have been loaded yet
+// This function loads pyodide as soon as the loadPyodide function is available
+async function ensurePyodide(){
+    while (typeof loadPyodide === 'undefined'){
+        await timer(30)
+    }
+    const py = await loadPyodide()
+    return py
+}
+
+async function initPyodide(){
+    const py = await ensurePyodide()
+    await py.loadPackage("micropip");
+    const micropip = await py.pyimport("micropip");
+
+    // Install Dependencies
+    await Promise.all([
+        micropip.install('hypothesis'),
+        micropip.install('pandas'),
+        micropip.install('numpy'),
+        micropip.install('pytest')
+    ])
+
+    const mountDir = ".";
+    await py.FS.mount(py.FS.filesystems.IDBFS, { root: "." }, mountDir);
+    await Promise.all([
+        py.FS.mkdir('/home/pyodide/challenges'),
+        py.FS.mkdir('/home/pyodide/userSolutions'),
+        py.FS.mkdir('/home/pyodide/tests')
+    ])
+    await py.FS.syncfs(true, function (err) {
+        console.log(err);
+        // handle callback
+    });
+    isPyodideReady.set(true)
+    return py;
+}
+
+// All functions which use pyodide have to resolve this promise to get the pyodide object
+// The Code inside the promise (loading an initializing pyodide) only runs once
+export const pyodidePromise = initPyodide()
```

Comment:
Do we need to export this? As far as I can see we only use this in the python-runner, and should not call it from elsewhere